feat: add devbind tool

mkfsn opened the pull request.

gshimansky committed:
Added KNI, port forwarding and full ARP support to NAT

- Cosmetic code refactoring into smaller separate functions
- Implemented KNI and port forwarding config
- Added creation of KNI devices and flows (no traffic forwarded yet)
- Implemented port forwarding (including KNI). It works to ssh from public interface.
- Direct all ARP and ICMP traffic to KNI if it is present on interface
- Fixed handling of ICMP and directing all ICMP traffic to KNI if it is present
- Fixed directing traffic to KNI in private subnet
- Implemented ARP requests when target destination MAC is unknown
- Fixed getting ARP response when KNI is enabled on interface
- Fixed directing traffic to KNI when it originates from NAT host
- Fixed printing IPv4 packet address and header
- Switched configuration to multiple hosts in local subnet
- Fixed bug with forwarded port reuse

gshimansky committed:
Implemented GRPC control over dump, subnet address and port forwarding

gshimansky committed:
Implemented DHCP client functionality to acquire IPv4 address

mkfsn committed:
(#1) move devices package from crimea to nff-go

mkfsn committed:
(#1) fix: make StringBuilder to be an unexported type

mkfsn committed:
(#1) fix: make newDevice to be an unexported function

mkfsn committed:
(#1) fix: lint

mkfsn committed:
(#1) refactor

mkfsn committed:
(#1) fix: make all stringbuilders to be unexported

mkfsn committed:
(#1) fix: lint - add comments

mkfsn committed:
(#1) refactor: make pciDevice to be unexported and add comments

mkfsn committed:
(#1) refactor: make vmbusDevice to be unexported

mkfsn committed:
(#1) fix: add comments

marcusschiesser reviewed (changes requested):
add `examples/devbind` with readme

gshimansky committed:
Merge pull request #465 from intel-go/gregory/smartnat

Added KNI, port forwarding and full ARP support to NAT

ifilippov committed:
Scheduler fix

Removed situation with zero division from measue function.
THis function itself can be removed for 210nn per 32 packets.

gshimansky committed:
Merge pull request #472 from intel-go/gregory/smartnat

Implemented GRPC control over dump, subnet address and port forwarding

gshimansky committed:
Trivial version bump for fedora image

ifilippov committed:
Merge pull request #473 from intel-go/ifilippov/fix

Scheduler fix

gshimansky committed:
Merge pull request #474 from intel-go/gregory/quickfix

Trivial version bump for fedora image

mkfsn committed:
(#1) feat: add a devbind example

Commit:
Updated comments

gshimansky committed:
Fixed docker images build

gshimansky committed:
Merge pull request #477 from intel-go/gregory/quickfix

Fixed docker images build

gshimansky committed:
Merge pull request #475 from intel-go/gregory/smartnat

Implemented DHCP client functionality to acquire IPv4 address

ifilippov committed:
Only supported RSS types

ifilippov committed:
Merge pull request #479 from intel-go/ifilippov/fix

Only supported RSS types

marcusschiesser committed:
(#1) added missing comments and package headers
